.github/workflows/build.yml

```diff
@@ -2,34 +2,66 @@ name: Build
 
 on:
   push:
-    branches:
-    - main
+    branches: [main]
+    paths:
+      - grammar.js
+      - src/**
+      - test/**
+      - bindings/**
+      - binding.gyp
   pull_request:
-  workflow_dispatch:
+    paths:
+      - grammar.js
+      - src/**
+      - test/**
+      - bindings/**
+      - binding.gyp
+
+concurrency:
+  group: ${{github.workflow}}-${{github.ref}}
+  cancel-in-progress: true
 
 jobs:
-  build:
+  test:
+    name: Test parser
+    runs-on: ${{matrix.os}}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-14]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Set up tree-sitter
+        uses: tree-sitter/setup-action/cli@v1
+      - name: Run parser and binding tests
+        uses: tree-sitter/parser-test-action@v2
+        with:
+          test-rust: true
+          generate: ${{runner.os == 'Linux'}}
+      - name: Output generated file size
+        if: runner.os == 'Linux'
+        run: du -sh src/*.json src/parser.c | sort -rh
+      - name: Parse sample files
+        uses: tree-sitter/parse-action@v4
+        with:
+          files: examples/Logger.kt
+  fuzz:
+    name: Fuzz scanner
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - name: Set up Node
-        uses: actions/setup-node@v3
+      - name: Checkout repository
+        uses: actions/checkout@v4
         with:
-          node-version: '18'
-      - name: Install dependencies
-        run: npm install
-      - name: Generate parser
-        run: npm run generate
-      - name: Verify that generated parser matches the repository
-        run: |
-          diff=`git diff HEAD -- src`
-          echo "$diff"
-          test -z "$diff"
```

Comment:
This also removes the verification check as far as I can tell. The rationale for including it can be found in #92. I'd be happy to discuss alternatives, but would like keep some check in place to avoid merging PRs containing potentially untrusted source code that is infeasible to review due to the size of the parser.

Reply:
`parser-test-action` verifies it with `generate: ${{runner.os == 'Linux'}}`

Reply:
Ah okay, thanks for clarifying! Does it reject incorrect parsers? It looks like CI will still pass even if that step fails:

https://github.com/tree-sitter/parser-test-action/blob/a573907d0859b4de355b1dfe0ef6f09126848755/action.yml#L54C7-L54C30

Reply:
You can check the log to see if it failed, but I can remove that.

Reply:
Yes, I'd like the workflow to fail if the check fails, otherwise this will likely go unnoticed.